verticapy/tests_new/plotting/base_test_files.py

```diff
@@ -0,0 +1,2256 @@
+        return obj.get_title()
+    if isinstance(obj, go.Figure):
+        return obj.layout["title"]["text"]
+    if isinstance(obj, Highchart):
+        return obj.options["title"].text
+    return None
+
+
+class BasicPlotTests:
+    """
+    Basic Tests for all plots
+    """
+
+    cols = []
+
+    # @property
+    @abstractmethod
+    def create_plot(self):
+        """
+        Abstract method to create the plot
+        """
+
+    @property
+    def result(self):
+        """
+        Create the plot
+        """
+        func, arg = self.create_plot()
+        return func(**arg)
+
+    def test_properties_output_type(self, plotting_library_object):
+        """
+        Test if correct object created
+        """
+        assert isinstance(self.result, plotting_library_object), "wrong object crated"
+
+    def test_properties_xaxis_label(self):
+        """
+        Testing x-axis label
+        """
+        test_title = self.cols[0]
+        assert get_xaxis_label(self.result) == test_title, "X axis label incorrect"
+
+    def test_properties_yaxis_label(self):
+        """
+        Testing y-axis title
+        """
+        test_title = self.cols[1]
+        assert get_yaxis_label(self.result) == test_title, "Y axis label incorrect"
+
+    def test_properties_zaxis_label(self):
+        """
+        Testing y-axis title
+        """
+        if len(self.cols) > 2:
+            test_title = self.cols[2]
+            assert get_zaxis_label(self.result) == test_title, "Z axis label incorrect"
+        else:
+            pass
+
+    def test_additional_options_custom_width_and_height(self):
+        """
+        Test custom width and height
+        """
+        func, arg = self.create_plot()
+        custom_width = 300
+        custom_height = 400
+        result = func(width=custom_width, height=custom_height, **arg)
+        assert (
+            get_width(result) == custom_width and get_height(result) == custom_height
+        ), "Custom width or height not working"
+
+
+class ACFPlot(BasicPlotTests):
+    """
+    Testing different attributes of ACF plot on a vDataFrame
+    """
+
+    @pytest.fixture(autouse=True)
+    def data(self, amazon_vd):
+        """
+        Load test data
+        """
+        self.data = amazon_vd
+
+    @property
+    def cols(self):
+        """
+        Store labels for X,Y,Z axis to check.
+        """
+        return ["lag", "value"]
+
+    def create_plot(self):
+        """
+        Create the plot
+        """
+        return (
+            self.data.acf,
+            {
+                "ts": "date",
+                "column": "number",
+                "p": 12,
+                "by": ["state"],
+                "unit": "month",
+                "method": "spearman",
+            },
+        )
+
+
+class VDCBarPlot(BasicPlotTests):
+    """
+    Testing different attributes of Bar plot on a vDataColumn
+    """
+
+    # Testing variables
+    COL_NAME = "check 2"
+    COL_NAME_2 = "check 1"
```

Comment:
This `COL_NAME_2` variable is not used. Doesn't Pylint complain about defined variables that are not used?

Reply:
it should but it is not for this case... not sure why.